### ISSUE TITLE
What is the purpose of the local symbol applicable accessibility ?

### ISSUE THREAD

**Issue by tonyhallett** [MicrosoftDocs/visualstudio-docs#2864]
I can see in [Roslyn](https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser_SymbolSpec.cs) that it corresponds to Accessibility.NotApplicable.  Is this the behaviour when omitting applicable_accessibilities ? If so should it be in the docs ?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b402d2e0-a53a-b61a-4da8-b5ee766d9301
* Version Independent ID: 612695df-5934-d967-d709-500ce02b1395
* Content: [.NET Naming Conventions For EditorConfig files - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-naming-conventions?view=vs-2017#feedback)
* Content Source: [docs/ide/editorconfig-naming-conventions.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/ide/editorconfig-naming-conventions.md)
* Product: **visual-studio-windows**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

**Comment by gewarren** [comment]
@sharwell Can you clarify what an applicable_accessibilities value of `local` should be used for in a naming style?

**Comment by sharwell** [comment]
It's used for items defined within a method, where accessibility cannot be specified. For local variables and local functions, the new accessibility doesn't really make a difference, but for local _constants_ it means users can separate the naming conventions of constants defined in a type from constants defined in a method.

```csharp
class TypeName
{
  const int X = 3; // This constant is defined in the type

  void Method()
  {
    const int Y = 4; // This constant is defined in a method, with local accessibility
  }
}
```

**Comment by tonyhallett** [comment]
I don't think that this is the best example of a use case for accessibility local as it is possible to differentiate using applicable symbol kinds instead.

It is important to clarify what symbols have local accessibility and under what circumstances  so that developers can target appropriately.  For instance the symbols for a local variable and local function are always local accessibility ( local function has private accessibility in Rolsyn but gets changed to local ).  Symbols for parameters and parameter types in the context of editor config do have accessibility levels that come from their containing symbol and as such can have local, public etc .  see here  https://github.com/dotnet/roslyn/blob/5b05de681c46ae58c9d5e05b85bd736c4986cd3c/src/Workspaces/Core/Portable/NamingStyles/Serialization/AccessibilityExtensions.cs#L25. This is contrary to what this document states '... For example, parameters do not have accessibility levels.'

The document is also incorrect in stating 'If you specify an accessibility level for a parameter naming convention, your naming rule will not function correctly.'  It is possible to apply one rule to parameters in a public method and a different rule to parameters in an anonymous function.

he document is incorrect here too: 'Do not specify an accessibility level as part of your naming convention if accessibility is not applicable to the kind of symbol you are targeting'.   Although this works as the default behaviour is * ( which is not mentioned in the documentation ) it is not necessary to do so.  It may also result in developers writing multiple rules that are not necessary.

Lastly developers might want to know that a local function symbol is static for purpose of modifier matching.  See here https://github.com/dotnet/roslyn/issues/27719

**Comment by sharwell** [comment]
> Lastly developers might want to know that a local function symbol is static for purpose of modifier matching.

This will have to change, as C# 8 introduces the notion of a static local function which is different from a non-static local function.

**Comment by gewarren** [comment]
I've updated the doc and it will go live tomorrow morning. I used Sam's example of 'local' accessibility, but if you have a better example please let me know. I guess you are saying that a naming convention for a local constant could be specified in the following way, without an accessibility?

applicable_kinds = local
required_modifiers = const
